### PR TITLE
[ONNX] Fix scaled_dot_product_attention with float scale

### LINKED ISSUE #125158: `scale` parsed as `float` in ONNX `scaled_dot_product_attention` implementation
Hi,
I have been getting errors looking like the one below when trying to export a model to ONNX within which I manually provide a `scale` argument to the scaled dot product attention calls:
```
   File "/usr/local/lib/python3.10/dist-packages/torch/onnx/symbolic_opset14.py", line 176, in scaled_dot_product_attention
    query_scaled = g.op("Mul", query, g.op("Sqrt", scale))
   File "<@beartype(torch.onnx._internal.jit_utils.GraphContext.op) at 0x7f10f6b3feb0>", line 44, in op
 beartype.roar.BeartypeCallHintParamViolation: Method torch.onnx._internal.jit_utils.GraphContext.op() parameter raw_args=0.03125 violates type hint typing.Union[torch.Tensor, torch.Value], as float 0.03125 not <protocol "torch.Tensor"> or <class "torch.Value">.
```

Looking at `symbolic_opset14.py`, I see that `_maybe_get_const` parses `scale` as a `float` which triggers the error as the type is not correct, cf.
[here](https://github.com/pytorch/pytorch/blob/da44d2f7fb805fe83803b79ccbd783494ec31d6b/torch/onnx/symbolic_opset14.py#L156)

Is this intentional? If I remove this line, then I manage to export successfully my model.
Thanks

### PR DESCRIPTION
**Pull request by titaiwangms**
Fixes #125158

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]

File: torch/onnx/symbolic_opset14.py [hunk 1]
```diff
@@ -150,7 +150,6 @@ def scaled_dot_product_attention(
     ), "is_causal and attn_mask cannot be set at the same time"
     assert not enable_gqa, "conversion of scaled_dot_product_attention not implemented if enable_gqa is True"
 
-    scale = symbolic_helper._maybe_get_const(scale, "f")
```
Comment (justinchuby) [inline torch/onnx/symbolic_opset14.py:153]: Just making sure: does it work for fp16?

Comment (borisfom) [reply torch/onnx/symbolic_opset14.py:153]: The fix seem to work with my FP16 case, I see no warnings either.

**Comment by justinchuby** [comment]
@titaiwangms did you want to cherry pick this as well? I can go under category (2) or (3) I think.

**Comment by titaiwangms** [comment]
> @titaiwangms did you want to cherry pick this as well? I can go under category (2) or (3) I think.

Sure! Looks like people need this.

**Pull request closed without merging**